app/views/layouts/_header.html.erb

```diff
@@ -2,6 +2,20 @@
 <% mobile_logo_path = SiteSetting['MobileLogoPath'] %>
 <% sticky_header = user_preference('sticky_header', community: false) == 'true' %>
 
+<script>
+  $(document).ready(function () {
+      $('a[data-header-slide="#search-slide"]').on('click', focusSearch);
+  });
+
+  var search = false;
+  function focusSearch() {
```

Comment:
A couple of notes:

- avoid using `var` at all costs, `const` for unchangeable cariables and `let` for reassignable is a must:
- the `focusSearch` function has a "scope bleed" - it should be called something like `toggleSearchFocus` and accept a single boolean parameter, which you then pass in the handler function. I'll provide example implementation a bit later